# Analysis; US organic Suppliers under NOP certification updated 2021

## From cleaning data to the analysis

Yira Rosas

The dataset for this analysis was taken of US Goberment page: https://organic.ams.usda.gov/integrity/Reports/DataHistory.aspx

To read the file you need saved it as "CSV UTF-8"

In [1]:
import pandas as pd
data = pd.read_csv("Organic_suppliers_utf.csv")
print(data.head(10))

C:\Users\Yira\Anaconda\lib\site-packages\IPython\core\interactiveshell.py:3444: DtypeWarning: Columns (3,15,20,25,30,38,46,53,58,59,62) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


                                      Certifier Name       Certifier Website  \
0                                           Required                Required   
1  More information about Accredited Certifying A...  From Certifier Profile   
2                                           Ekoagros         www.ekoagros.lt   
3                     Transitioning to New Certifier                     NaN   
4                                           Ekoagros         www.ekoagros.lt   
5                                           Ekoagros         www.ekoagros.lt   
6                                           Ekoagros         www.ekoagros.lt   
7               Organic Crop Improvement Association            www.ocia.org   
8                                           Ekoagros         www.ekoagros.lt   
9            Ecological Farming Control Organization            www.etko.org   

  Certifier Email Address                                       Operation ID  \
0                Required              

## Staring the cleaning

Ok, the first problem that we can see, it is the DtypeWarning. This basically means that our dataset has columns with multiple dtypes (like a mix of str and int) Pandas gets confused and it needs we tells it what to do.

In [2]:
#Defining data type of columns with multiple data type

types= {'Operation ID': 'str','Additional Certified Products Under CROPS Scope ':'str',
         'Additional Certified Products Under LIVESTOCK Scope':'str',
         'Additional Certified Products Under WILD CROPS Scope':'str',
         'Additional Certified Products Under HANDLING Scope':'str',
         'County Code':'str',
         'Community Supported Agriculture (CSA)':'str',
         'Restaurant':'str',
         'Retail Food Establishment':'str',
         'Slaughterhouse':'str'}

#Reading data with the new data type

newdata= pd.read_csv("Organic_suppliers_utf.csv",dtype=types)

print(newdata.head())

                                      Certifier Name       Certifier Website  \
0                                           Required                Required   
1  More information about Accredited Certifying A...  From Certifier Profile   
2                                           Ekoagros         www.ekoagros.lt   
3                     Transitioning to New Certifier                     NaN   
4                                           Ekoagros         www.ekoagros.lt   

  Certifier Email Address                                       Operation ID  \
0                Required                                           Required   
1  From Certifier Profile  NOP's 10-digit unique ID for operation.  First...   
2    ekoagros@ekoagros.lt                                         6905257152   
3                     NaN                                         0007848893   
4    ekoagros@ekoagros.lt                                         6903275443   

               Operation Name  \
0    

In [3]:
# Getting the info about the dataset
print(newdata.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66834 entries, 0 to 66833
Data columns (total 68 columns):
 #   Column                                                             Non-Null Count  Dtype  
---  ------                                                             --------------  -----  
 0   Certifier Name                                                     66833 non-null  object 
 1   Certifier Website                                                  65234 non-null  object 
 2   Certifier Email Address                                            65234 non-null  object 
 3   Operation ID                                                       66833 non-null  object 
 4   Operation Name                                                     66833 non-null  object 
 5   Other/Former Names                                                 9025 non-null   object 
 6   Client ID                                                          51435 non-null  object 
 7   Contact First Name    

## Dealing with the nulls

We have a lot of columns and rows with empty info, we need to think what to do with them.

In [4]:
# Let's see which columns have the most null values and sort them.
print(newdata.isnull().sum().sort_values(ascending = False))

Unnamed: 67                                             66834
Additional Certified Products Under WILD CROPS Scope    66833
Additional Certified Products Under LIVESTOCK Scope     66833
Additional Certified Products Under CROPS Scope         66832
Restaurant                                              66830
                                                        ...  
Operation Certification Status                              1
Operation Name                                              1
Operation ID                                                1
Data as of Date                                             1
Certifier Name                                              1
Length: 68, dtype: int64


We have columns with 66,830 nulls or more, this is almost all column empty because we have 66830 rows, for this excersice I will delete them.

In [5]:
#First we will create a sub-dataframe with our original data, this to return to our original data in case we want

#New sub-dataframe
suppliers=newdata

#Dropping columns with 80% or more nulls
delete_nulls= len(suppliers) * .4
suppliers.dropna(thresh = delete_nulls, axis = 1, inplace = True)

print(suppliers.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66834 entries, 0 to 66833
Data columns (total 26 columns):
 #   Column                                Non-Null Count  Dtype 
---  ------                                --------------  ----- 
 0   Certifier Name                        66833 non-null  object
 1   Certifier Website                     65234 non-null  object
 2   Certifier Email Address               65234 non-null  object
 3   Operation ID                          66833 non-null  object
 4   Operation Name                        66833 non-null  object
 5   Client ID                             51435 non-null  object
 6   Contact First Name                    40649 non-null  object
 7   Contact Last Name                     40546 non-null  object
 8   Operation Certification Status        66833 non-null  object
 9   Effective Date of Operation Status    66833 non-null  object
 10  CROPS Scope Certification Status      39552 non-null  object
 11  Effective Date of CROPS Stat